src/controllers/filters/lang/Filter.hpp

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include "controllers/filters/lang/expressions/Expression.hpp"
+#include "controllers/filters/lang/Types.hpp"
+
+#include <QString>
+
+#include <memory>
+#include <variant>
+
+namespace chatterino {
+
+class Channel;
+struct Message;
+using MessagePtr = std::shared_ptr<const Message>;
+
+}  // namespace chatterino
+
+namespace chatterino::filters {
+
+static const QMap<QString, Type> messageTypingContext = {
```

Comment:
I'd prefer if this is `MESSAGE_TYPING_CONTEXT` too (rare clang-tidy agree), alongside a small comment explaining this